File: android/portal-socket/android/src/main/java/ar/portal659/socket/BootReceiver.java
package ar.portal659.socket;

import android.content.BroadcastReceiver;
import android.content.Context;
import android.content.Intent;
import android.os.Build;

/**
 * Levanta el foreground service al encender el celular: la notificación fija vuelve
 * a aparecer sin tocar nada (el WS se reconecta al abrir la app desde la notificación).
 */
public class BootReceiver extends BroadcastReceiver {
    @Override
    public void onReceive(Context context, Intent intent) {
        if (intent == null || intent.getAction() == null) return;
        String action = intent.getAction();
        if (!Intent.ACTION_BOOT_COMPLETED.equals(action) && !"android.intent.action.QUICKBOOT_POWERON".equals(action)) {
            return;
        }
        Intent svc = new Intent(context, PortalPrintService.class);
        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.O) {
            context.startForegroundService(svc);
        } else {
            context.startService(svc);
        }
    }
}

File: portal-wa-link/app/src/main/java/ar/portal659/walink/BootReceiver.kt
package ar.portal659.walink

import android.content.BroadcastReceiver
import android.content.Context
import android.content.Intent

/** Arranca el relay al encender el teléfono (solo si el usuario lo tiene activo). */
class BootReceiver : BroadcastReceiver() {
    override fun onReceive(context: Context, intent: Intent) {
        if (intent.action == Intent.ACTION_BOOT_COMPLETED && Config.enabled(context)) {
            RelayService.start(context)
        }
    }
}